src/ui/zcl_abapgit_gui_view_repo.clas.abap

```diff
@@ -360,6 +368,59 @@ CLASS zcl_abapgit_gui_view_repo IMPLEMENTATION.
   ENDMETHOD.
 
 
+  METHOD open_in_master_language.
+
+    CONSTANTS:
+      lc_abapgit_tcode TYPE tcode VALUE `ZABAPGIT` ##NO_TEXT.
+
+    DATA:
+      lv_master_language TYPE spras,
+      lt_spagpa          TYPE STANDARD TABLE OF rfc_spagpa,
+      ls_spagpa          LIKE LINE OF lt_spagpa,
+      ls_item            TYPE zif_abapgit_definitions=>ty_item.
+
+    " https://blogs.sap.com/2017/01/13/logon-language-sy-langu-and-rfc/
+
+    lv_master_language = mo_repo->get_dot_abapgit( )->get_master_language( ).
+
+    IF lv_master_language = sy-langu.
+      zcx_abapgit_exception=>raise( |Repo already opened in master language| ).
+    ENDIF.
+
+    ls_item-obj_name = lc_abapgit_tcode.
+    ls_item-obj_type = |TRAN|.
+
+    IF zcl_abapgit_objects=>exists( ls_item ) = abap_false.
+      zcx_abapgit_exception=>raise( |Please install the abapGit repository| ).
+    ENDIF.
+
+    SET LOCALE LANGUAGE lv_master_language.
+
+    ls_spagpa-parid  = zif_abapgit_definitions=>c_spagpa_param_repo_key.
+    ls_spagpa-parval = mo_repo->get_key( ).
+    INSERT ls_spagpa INTO TABLE lt_spagpa.
+
+    CALL FUNCTION 'ABAP4_CALL_TRANSACTION'
+      DESTINATION 'NONE'
+      STARTING NEW TASK 'ABAPGIT'
+      EXPORTING
+        tcode                   = lc_abapgit_tcode
+      TABLES
+        spagpa_tab              = lt_spagpa
+      EXCEPTIONS
+        call_transaction_denied = 1
+        tcode_invalid           = 2
+        communication_failure   = 3
+        system_failure          = 4
+        OTHERS                  = 5.
+
+    IF sy-subrc <> 0.
+      zcx_abapgit_exception=>raise( |Error from ABAP4_CALL_TRANSACTION. Subrc = { sy-subrc }| ).
+    ENDIF.
+
+  ENDMETHOD.
```

Comment:
After this you have two abapGit sessions open. It might be a good idea to add a `LEAVE PROGRAM` here?

Reply:
I have mixed feelings about that. In most cases I'd prefer to keep the old session, because e.g. I just want to pull the latest abapGit in EN and the continue work in DE. Maybe we should ask the user whether he wants to replace the session or open a new one?

Reply:
Hmm, I agree. What do you think of a MESSAGE TYPE 'S' in the "normal" abapGit session that indicates what happened so it's clear for the user?

Reply:
Added success message